lib/core/errors/failures.dart

```diff
@@ -9,9 +12,21 @@ abstract class Failure extends Equatable {
   List<Object?> get props => [reason];
 }
 
-// General failures
 class ServerFailure extends Failure {
   const ServerFailure(super.reason);
+
+  factory ServerFailure.fromResponse(Response response) {
+    try {
+      final jsonString =
+          json.decode(response.bodyString) as Map<String, dynamic>;
+
+      final message = jsonString['message'] as String?;
+
+      return ServerFailure(message ?? response.bodyString);
```

Comment:
At our last meeting, we discussed changing this, so we wouldn't show the bodyString to the user if the message was null. Instead, we would show some default message like "An unknown error occurred", and then log the bodyString to firebase, so we could actually act on whatever error occurred. I think that is something we should go ahead and do, so that we get actionable data for ourselves, and the user does not see bogus info.